### DISCUSSION TITLE
251

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#529] [Comments]
http://jojoldu.tistory.com/251

**Comment by jojoldu** [comment]
작성자: 허니몬

^^ 깔끔하게 정리된 멋진 글이네요.
잘보고 갑니다!

**Comment by jojoldu** [comment]
작성자: 원스

좋은 글 잘 보고 갑니다.^^
고생 하셨네요~
앞으로도 좋은 글 부탁 드립니다.

새해 복 많이 받으셔요~!!

**Comment by jojoldu** [comment]
작성자: soulduse@naver.com

http://localhost:8080/h2-console 입력 후,
JDBC Url 부분에 jdbc:h2:mem:testdb를 꼭 넣어줘야 제대로 값이 반영되네요 
가장 처음 열었을때 기본값 jdbc:h2:~/test 이었나..?
상관없는줄알고 기본값으로 connect 했다가, 테이블 생성이 안되서 한창 삽질했네요 ^^;;; 
혹시 저 같은 분이 계실까봐 메모해둡니다!

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

앗 ㅋㅋㅋ 고마워 현우야!

**Comment by jojoldu** [comment]
작성자: 떱스턴

한창 하다가 안되서 찾았는데
여기에 이런 답변이 있었네요 
감사합니다!!
아 그리고 포스팅 잘 보고있습니다.
감사합니다!!

**Comment by jojoldu** [comment]
작성자: betterfly

많은 기술 블로그를 보면서 공부하지만
이렇게 잘 정리되고, 꼼꼼하게 설명해주시는 곳을 본적이 없었던 것 같습니다
깃으로 취업정보도 구하고, 스프링부트까지 
정말 너무 감사히 도움을 구하고있습니다
감사드립니다
(저도 디비에서 테이블이 보이질 않아 삽질하고있었는데 윗분 댓글보고 해결했네요 감사합니다 ㅎㅎ)

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

고맙습니다^^ 좀더 좋은 내용으로 계속 추가하겠습니다.
앞으로도 잘 부탁드리겠습니다!

**Comment by jojoldu** [comment]
작성자: 빌더제발


안녕하세요 정말 좋은 글 감사힙니다. 질문이 있어서 댓글답니다 
Error:(19, 22) java: The method builder() is undefined for the type com.tutorial2.ohs.domain.posts.Posts

위와 똑같이 코드를 돌려도 builder()가 없다는 말이 나오는데요, 혹시 무슨 상황인지 짐작이 가시는지요

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

안녕하세요!
롬복 문제인것 같은데요
이클립스를 쓰시는것 같은데 이클립스와 롬복 버전을 최신으로 맞춰보시겠어요?
본문에 있는 링크(http://countryxide.tistory.com/16) 에 롬복 설치법이 있긴 한데 최신 버전이 아니라서 설치방법만 참고하시면 될것 같아요.

참고로 스택오버플로우에서 발견한 동일 이슈(
https://stackoverflow.com/questions/26189167/projectlombok-method-undefined-in-eclipse-after-installing-eclipse-luna)

**Comment by jojoldu** [comment]
작성자: 윤찬명

안녕하세요. 스프링부트를 처음 접하고 쭉 따라하고있습니다. Intellij + Gradle 환경에서 진행하고 있는데요.
 윗분과 똑같은 Builder 오류가 발생해 Lombok을 업데이트하고나니 (http://countryxide.tistory.com/32) 빨간줄은 사라졌는데, 실행하면 cannot find symbol symbol: method builder() Error가 발생하고있습니다. 도움주실 수 있을까요?

**Comment by jojoldu** [comment]
작성자: 윤찬명

글은 정말 잘 보고있습니다. 감사합니다 :)

**Comment by jojoldu** [comment]
작성자: 윤찬명

앗!! 해결했습니다. 
Intellij에서 Annotation Processors를 Enable해주면 되네요! 괜히 Intellij 버전업하고 생쇼한.. ㅎㅎ;; 감사합니다. 잘 따라 배우겠습니다.

**Comment by FAILtURE** [comment]
2-2 계속 fail만 뜹니다.. 내용이 뭔가 부족한것 같아요
Enable Annotation Processors도 하였습니다.
빨간 글이 아주 많이 뜨는데 가장 상위가 java.lang.IllegalStateException: Failed to load ApplicationContext입니다.. 뭐가 문제일까요..

Spring Boot :: (v2.0.0.RELEASE)\
JDK 9 다운받고 8로 프로젝트 똑같이 만들었어요.

**Comment by jojoldu** [comment]
음ㅠㅠ 롬북은 자바9를 지원하지않습니다ㅠㅠ 자바8로 변경하셔야됩니다ㅠ

**Comment by sskim91** [comment]
안녕하세요 창천향로님 다름아니라 해당 포스트에 제가 예전에 질문 댓글을 달아놓은게 있었는데 다시 와서 보려고 하니까
댓글서비스를 utteranc(?) 로 변경 하셨는지 보이지가 않네요ㅠㅠ 실제로 티스토리 카테고리에는 댓글 개수가 (24)로 카운트 되어있는데 여기 댓글 코멘트 개수는 14로 카운트가 되어있는것 같습니다. 혹시 댓글이 다 사라져서 이제 볼수 없는건가요??

**Comment by jojoldu** [comment]
@sskim91 음 방금 댓글 노출을 다시 활성화시켰는데요! 혹시 보이시나요?
티스토리 API에서 비공개 댓글은 제공을 안해줘서 비공개 댓글까지는 마이그레이션이 되지 않았네요 ㅠ
금일 (2018-03-14)까지는 기존 댓글 오픈해놓겠습니다
혹시 댓글 남기신게 계속 필요하시면 깃헙 댓글로 남겨주세욤!

**Comment by sskim91** [comment]
@jojoldu 
떱스턴 이라는 아이디로 댓글을 작성했는데 아래에 보이네요!! 
정말 감사합니다.!! 
음 댓글 작성할 때에  비공개? 이런건 안한거 같은데 왜 안나왔는지는 모르겠네요. 
댓글 남긴게 보였으면 좋겠는데 어떻게하면 좋을까요??

**Comment by jojoldu** [comment]
@sskim91 음 젤 편한 방법은 동일한 내용을 여기 깃헙댓글로 그대로 다시 남기시는게 좋지 않을까 싶습니다!
댓글 복사해서 남겨주시면, 제 댓글도 그 아래에 추가해놓겠습니다!

**Comment by sskim91** [comment]
@jojoldu
네 ㅎㅎ 감사합니다. 부끄럽지만 다시 한번 복붙 하겠습니다. 좋은하루 보내세요!!
-------------------
질문있습니다..
바보같은 질문 같아도 초보자라 이해해주세요
본문에
private PostsRepository postsRepository; 
public WebRestController(PostsRepository postsRepository) { 
this.postsRepository = postsRepository; 
}
이 부분에서 bean을 주입받는 방법중에 3가지에서 생성자로 주입받는 방법이라고 하셨는데
그럼 WebRestController 라는 객체는 어디서 생성하는건가요?? 코드상에서 없어서 스프링이 알아서 해주는것인가요?ㅠㅠ
postRepostory라는 파라미터를 어디서 받는건지 궁금합니다.

**Comment by jojoldu** [comment]
@sskim91 넵넵 저도 아래 남겨놓겠습니다^^ 감사합니다!

안녕하세요! 질문 감사합니다!
말씀하신대로 흔히 얘기하는 스프링빈 (@Controller, @Service, @Repository, @Bean, @Component) 등은 스프링의 Bean Factory에서 해당 빈들을 중앙관리소 형태로 관히합니다.
스프링빈의 생명주기를 관리하고 (생성 등) 의존성들(예: A 컨트롤러가 a 서비스가 필요한 상황)을 대신 넣어주는 역할을 합니다.

이전에 작성한 글을 참고하시면 더 좋습니다!
[http://jojoldu.tistory.com/28](http://jojoldu.tistory.com/28)

**Comment by alsdns016** [comment]
POST 요청 시 200 성공은 뜨는데 H2의 어느 디비에 생성되는지 모르겠습니다.

posts테이블을 저가 생성해줘야하는건가요? 아니면 자동으로 (테스트를 돌리거나 postman으로 POST 요청 시) 생성이 되는건가요?ㅠㅠ

H2-console의 기본 값 'test'나 'testdb'에 가도 해당 테이블은 없습니다 ㅠ

(임의로 posts를 생성해도 200성공은 뜨는데... 로우가 생겨나지않아요 ! ㅠㅠ,...)

**Comment by alsdns016** [comment]
윽 위에 댓글을 지우지못해 다시 남깁니다.

H2의 JDBCUrl이 그냥  'testdb'가아니라 

JDBCUrl => 'jdbc:h2:mem:testdb' 이거군요..!

**Comment by jojoldu** [comment]
@alsdns016 넵넵 ㅎㅎ; 본문에 설명이 빠졌네요 ㅠ 찾으셨어서 다행입니다 ㅠ

**Comment by ghost** [comment]
포스팅 잘 보고 가요.

구현하면서 번외로 유용한 정보도 많네요! 
상세한 설명 굿굿입다.

**Comment by joont92** [comment]
포스팅 잘봤습니다 ㅎㅎ 감사합니다.

혹시 BaseTimeEntity를 abstract class로 생성하신 이유가 있나요??

**Comment by jojoldu** [comment]
@joont92 해당 클래스만으로 인스턴스 생성되는걸 막기 위함입니다.
BaseTimeEntity는 시간 관련된 내용들을 담당하고, 어디까지나 엔티티 클래스들을 보조하기 위함인데,
이를 코드 상으로 표현하기엔 추상 클래스가 적당하기 때문에 선택하게 됐습니다 :)

**Comment by chh9975** [comment]
builder 에러나는 건 이렇게 저렇게 하다가 해결이 됐는데 
postman을 이용하여 데이터를 전송하면 webservice.web.WebRestController.savePosts(WebRestController.java:23) ~[main/:na]
이런 에러가 나네요... 500에러가 나면서.. 제 생각엔 builder가 제대로 해결이 안된 것 같은데
해결방법이 어떻게 되나요??

**Comment by gonghojin** [comment]
항상 많은 거 배우고 갑니다. si에서 시작하고 있는 신입이지만, 창천향로님과 같은 과정을 목표로 코딩 공부 열심히 하고 있습니다. si에 계셨던 경험을 바탕으로 서비스업체와의 차이를 설명해주시는 소소한 글 또한 정말 도움이 많이 되고있습니다. 좋은 글 감사합니다~

**Comment by SlowMonk** [comment]
일단 eclipse 에서 진행 하고 있는데요 
java.lang.TypeNotPresentException: Type org.springframework.test.context.junit4.SpringRunner not present
	at java.base/sun.reflect.annotation.TypeNotPresentExceptionProxy.generateException(TypeNotPresentExceptionProxy.java:46)
	at java.base/sun.reflect.annotation.AnnotationInvocationHandler.invoke(AnnotationInvocationHandler.java:86)
	at com.sun.proxy.$Proxy2.value(Unknown Source)
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:33)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestLoader.createUnfilteredTest(JUnit4TestLoader.java:87)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestLoader.createTest(JUnit4TestLoader.java:73)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestLoader.loadTests(JUnit4TestLoader.java:46)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:522)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:760)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:460)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:206)
Caused by: java.lang.ClassNotFoundException: org.springframework.test.context.junit4.SpringRunner
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:582)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:190)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:499)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:374)
	at java.base/sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:114)
	at java.base/sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:125)
	at java.base/sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
	at java.base/sun.reflect.annotation.AnnotationParser.parseSig(AnnotationParser.java:439)
	at java.base/sun.reflect.annotation.AnnotationParser.parseClassValue(AnnotationParser.java:420)
	at java.base/sun.reflect.annotation.AnnotationParser.parseMemberValue(AnnotationParser.java:349)
	at java.base/sun.reflect.annotation.AnnotationParser.parseAnnotation2(AnnotationParser.java:286)
	at java.base/sun.reflect.annotation.AnnotationParser.parseAnnotations2(AnnotationParser.java:120)
	at java.base/sun.reflect.annotation.AnnotationParser.parseAnnotations(AnnotationParser.java:72)
	at java.base/java.lang.Class.createAnnotationData(Class.java:3710)
	at java.base/java.lang.Class.annotationData(Class.java:3699)
	at java.base/java.lang.Class.getAnnotation(Class.java:3604)
	at org.junit.internal.builders.IgnoredBuilder.runnerForClass(IgnoredBuilder.java:10)
	... 11 more
이런 에러가 뜨네요 어떻게 해야 하나요

**Comment by n1tjrgns** [comment]
빨간 글이 아주 많이 뜨는데 가장 상위가 java.lang.IllegalStateException: Failed to load ApplicationContext입니다.. 뭐가 문제일까요..

위에서 이 댓글과 증상이 똑같습니다!
롬복이 자바 9를 지원하지 않는다그래서 확인해보니 자바 10을쓰고있었습니다.
하지만 자바8로 바꿨는데도 증상이 똑같이나오네요..

**Comment by choish1919** [comment]
와 정말 감사합니다
이정도면 돈주고 배울정돈데요!!

**Comment by joshua1988** [comment]
아 역시 동욱님입니다... 정말 오랜만에 동욱님 블로그 글을 정독했네요. 정리 깔끔하게 잘해주셔서 넘 감사드립니다 :) 도움 많이 되었어요!

**Comment by Y-k-Y** [comment]
가끔 와서 OKKY컨퍼런스 후기글을 읽었었는데 이런 강의 자료가 있는지는 몰랐네요.
정말 좋은 글 감사합니다👍

**Comment by junwoochoi** [comment]
정말 감사합니다!!

**Comment by osj4532** [comment]
안녕하세요 정말 잘보면서 따라하고 있습니다. 저는 윈도우 환경에서 이클립스로 실행중인데요. 포스트맨으로 데이터를 넘겨주는 상황에서 

2018-12-30 17:14:55.655  WARN 13276 --- [nio-8080-exec-9] .w.s.m.s.DefaultHandlerExceptionResolver : Resolved [org.springframework.web.HttpRequestMethodNotSupportedException: Request method 'POST' not supported]

이와 같은 오류 메시지가 뜨면서 데이터가 넘어가지 않는데 혹시 무슨 문제인지 알수 있을까요??

**Comment by dwlee2woowa** [comment]
@osj4532 음 오류 메세지를 보시면 이유가 나와있는데요
요청하신 주소는 POST 가 지정안되어있다고 되어있습니다.
Get만 만들어놓으신것 같은데, POST로 Postman에서 요청하신것 같습니다.
에러 보시면 아마 이해되실거에요

**Comment by osj4532** [comment]
혹시 몰라서 get으로 넘겼을때는 잘 넘어가는 걸 확인했습니다.

**Comment by osj4532** [comment]
@dwlee2woowa 답변 감사합니다. controller에 제가 get으로 해놓았더군요; 다음부터 자세히 보고 질문 하겠습니다

**Comment by jojoldu** [comment]
@osj4532 네^^ 수고하세요 :)

**Comment by newzinee** [comment]
글이 정리가 정말 잘 되어있네요! 따라하기도 쉽고 중간중간 제가 모르는 내용도 다 설명해주셔서, 또 검색해보는 일 없이 이 포스트만으로도 잘 이해가 되었습니다. 
좋은 글 감사드립니다.

**Comment by jojoldu** [comment]
@treasureBear94 와 칭찬 감사합니다 :)
다음에도 좋은 내용으로 포스팅 하겠습니다!
감사합니다 :)

**Comment by SimKiSeong** [comment]
Test할때 
assert함수를
import static org.springframework.test.util.AssertionErrors.assertTrue;
이렇게 스프링에 있는 것을 쓰면 오류가 나네요
import static org.junit.Assert.assertTrue;
이렇게 junit에 있는 것을 썻더니 잘 돌아갓네요 ㅎㅎ

**Comment by esp2ar0** [comment]
스프링을 이제 막 공부하고 있는 학생입니다.
NoArgsConstructor를
> Entity 클래스를 프로젝트 코드상에서 기본생성자로 생성하는 것은 막되, JPA에서 Entity 클래스를 생성하는것은 허용하기 위해 추가

라고 하셨는데, JPA에서 어떤 경우에 Entity 클래스를 생성하게 되는지 궁금합니다.

**Comment by JongHoo** [comment]
매번 잘 보고있습니다. 많은 도움을 얻고있습니다 ㅠㅠ
혹시 이 포스트의 내용으로 배운것들을 개인 블로그에 작성해도 괜찮을까요?
출처를 남겨야한다면 그렇게 하겠습니다!

**Comment by jojoldu** [comment]
@JongHoo 네 출처만 잘 보이게 해주시면 문제 없습니다 :)

**Comment by jojoldu** [comment]
@esp2ar0 대표적으로는 ```select```할때입니다.
JPA에서 DB 데이터를 select해올때 객체를 생성하는 것은 JPA인데요.
이때 기본 생성자가 없으면 생성되지 않아 실패합니다.

**Comment by codemakim** [comment]
진짜 잘 쓰셨다...

**Comment by funchcode** [comment]
블로그보면서 정말 감사하게 공부하고 있습니다!
좋은 글 감사합니다 :)

**Comment by VA-Kim** [comment]
안녕하세요!! 따라해보고 있는데, 막히는 부분이 있어서 질문드려요!

테스트 코드도 통과했는데, 실제로 서버 구동시키고 포스트맨으로 post 날려보면 400에러가 발생합니다. 구글링으로도 답을 찾지 못해서 문의드려요.ㅠㅠ 혹시나해서 동욱님 소스코드를 그대로 받아서 해봤는데, 소스의 문제는 아닌 것 같습니다. 동일한 에러가 발생 해서요.

"message": "Required request body is missing: public void com.jojoldu.webservice.web.WebRestController.savePosts(com.jojoldu.webservice.dto.posts.PostsSaveRequestDto)",

postman에서 raw에 json으로 설정하고 했는데.. 제가 사용을 못하는 건가요? ㅠㅠ

**Comment by newzinee** [comment]
안녕하세요~ 질문이 있어서 글 씁니다! 

중간에 
> Controller에서 @RequestBody로 외부에서 데이터를 받는 경우엔 기본생성자 + set메소드를 통해서만 값이 할당됩니다. 

라고 하셨는데, @Setter, @NoArgsConstructor없이 @Getter만 넣고, postman으로 테스트했을 때, 잘 동작하더라구요! 
ㅎㅎ 왜 그런걸까요..? 궁금해서 댓글 남깁니다!

**Comment by jojoldu** [comment]
@treasureBear94 좋은질문 감사합니다 :)
해당 내용은 제가 오해가 있었고,
자세한 설명은 아래 링크에 남겼으니 참고하시면 됩니다!

질문 감사합니다 :)

https://jojoldu.tistory.com/407

**Comment by ssqkrgustj** [comment]
안녕하세요 스프링 부트 초보의 질문입니다ㅠㅠ
PostRepositoryTest.java 파일의
 @Test
    public void 게시글저장_불러오기() {
        //given
        postsRepository.save(Posts.builder() 
부분에서, builder()를 Posts.java에 create 했을 때, 오류가 나는데요, 해결방법을 알 수 있을까요?ㅜ

create를 여러번 해 본 결과,
Posts.java
    public static <B> ChartBuilder<B> builder(){
    }
    public static DirectoryChooserBuilder builder() {
    }
이렇게 두 가지의 경우로 생성되는데요, 
ChartBuilder<B>와, DirectoryChooserBuilder 에 취소선이 생깁니다..

**Comment by ssqkrgustj** [comment]
세상에.. 오류를 무시한 채로 실행하니 되네요;;; 테스트라서 그런건지.. 해결된 것 같습니다ㅠㅠ 좋은 게시글 감사합니다!

**Comment by sdYoo** [comment]
테스트까지 실행이 되는데 H2 DB에는 POSTS 테이블 생성이 안되어 있습니다... 왜그런걸까요??ㅠ

**Comment by sdYoo** [comment]
위 댓글보고 h2 접속시 jdbc:h2:mem:testdb 해야만 되네요... 혹시 mem , testdb 는 어느시점에 생성되는 건가요? 그냥 디폴트 설정이 저렇게 잡히는 건가요?

**Comment by kdh0981** [comment]
너무 잘 정리된 글이네요~ JPA 이해하는데 큰 도움되었습니다.
감사합니다ㅎㅎ 하트도 누르고 갑니다.

**Comment by CEmmanuelP** [comment]
현재 application.yml설정 후 h2-console로 들어가고 싶은데, 안들어가지고 에러페이지로 빠지고 있습니다.
어디가 문제인지 파악이 되지 않아서 문의드리고 싶은데요. 한번 봐주실 수 있을까요?

github: https://github.com/CEmmanuelP/spring-webservice

**Comment by kuinsoo** [comment]
스프링 부트를 시작하기에 너무 막연했는데 .. 
정말 자세하게 좋은 자료 감사합니다. 
덕분에 깊이 있는 개발이 되는것 같아 뿌듯하네요.  천천히 착실히 배워서  나중에 생겨날 후배님들에게 다 전해 드려야겠네요 ^^

**Comment by moonsiri** [comment]
저는

**Comment by moonsiri** [comment]
저는 
IDE : IntelliJ IDEA Version 2019.1 Community
OS : Windows10
SpringBoot 2.1.8
Java : jdk 1.8
build Tool : Gradle

환경으로 개발중인데 


build.gradle 내용은 아래와 같아요.

ext['hibernate.version'] = '5.2.11.Final'  이 내용을 어떻게 추가하면 될까요?ㅜㅜ



plugins {
    id 'org.springframework.boot' version '2.1.8.RELEASE'
    id 'io.spring.dependency-management' version '1.0.8.RELEASE'
    id 'java'
}

group = 'com.demo'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = '1.8'

configurations {
    compileOnly {
        extendsFrom annotationProcessor
    }
}

repositories {
    mavenCentral()
}

dependencies {
    implementation 'org.springframework.boot:spring-boot-starter-actuator'
    implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
    implementation 'org.springframework.boot:spring-boot-starter-web'
    compileOnly 'org.projectlombok:lombok'
    runtimeOnly 'com.h2database:h2'
    annotationProcessor 'org.projectlombok:lombok'
    testImplementation 'org.springframework.boot:spring-boot-starter-test'
}

**Comment by jojoldu** [comment]
@moonsiri 님 안녕하세요 :)
스프링 부트 2 버전을 쓰신다면 추가안하셔도 됩니다 :)
이 시리즈는 스프링 부트 1.5 를 기준으로 하고 있어서 필요한 설정이구요 ^^;
2쓰신다면 괜찮습니다!

**Comment by moonsiri** [comment]
@jojoldu 아하 그렇군요! 감사합니다!!!! 글이 정말 좋아요ㅜㅜㅜ 스프링부트로 웹개발하는거 공부하고싶었는데 딱 원하는 게시글을 찾아서 열심히 따라 공부중입니다!

**Comment by jojoldu** [comment]
@moonsiri ㅎㅎㅎ;; 감사합니다!

**Comment by ProgWon** [comment]
PostRepositoryTest 부분에서 
@Autowired
PostsRepository postRepository 부분 에러가 나네요 ㅠㅠ

Could not autowire. No beans of 'PostRepository' type found.
Inspection info:Checks autowiring problems in a bean class.

에러 본문입니다 ㅠㅠ 뭐가 문제일까요??

**Comment by ProgWon** [comment]
@ckdrb7017

@Component, @Repository 둘다 작동을 안 하네요..... ㅠㅠ....

**Comment by jojoldu** [comment]
@ProgWon 님 해당 프로젝트를 Github 으로 올려서 링크를 주시면 좀 더 확인이 쉬울것 같습니다 :)
말씀하신 경우에 ```extends JpaRepository```가 선언된 인터페이스이면 ```@Repository```가 없어도 Bean으로 자동 등록이 되는데요
예상되는 문제는 2가지가 있는데요

1) ```@SpringBootApplication``` 이 선언된 클래스 (Application.java) 위치가 최상위 패키지가 아니다
2) 테스트 코드에 알수없는 문제가 있다

전체 코드를 제가 볼수있다면 좀 더 확인이 쉬울것 같습니다 :)

**Comment by kjd91** [comment]
잘 따라가다가 오류때문에 질문 올립니다..
게시글저장_불러오기 테스트에서 오류가 나오는데요.. 오류내용은 아래내용과 같고 github 링크입니다.

```

***************************
APPLICATION FAILED TO START
***************************

Description:

Failed to configure a DataSource: 'url' attribute is not specified and no embedded datasource could be configured.

Reason: Failed to determine a suitable driver class


Action:

Consider the following:
	If you want an embedded database (H2, HSQL or Derby), please put it on the classpath.
	If you have database settings to be loaded from a particular profile you may need to activate it (no profiles are currently active).


```

https://github.com/kjd91/spring-webservice

**Comment by Guafindel** [comment]
https://github.com/Guafindel/SpringBoot/tree/Boot_Guifindel/Spring-WebService

제 github 주소입니다. h2-console과 postman모두 통과하고 제대로된 결과값이 나오는 것을 확인했습니다.
그러나 JUnit 검사결과에서는 알 수 없는 에러가 하나 발생하고있으며 BaseTimeEntity 경우에는 Failures가 하나 발생합니다.

assertTrue(posts.getCreatedDate().isAfter(now)); 
오류를 추적해보니 이 코드에서 오류가 나오는데 왜 이러는지는 모르겠습니다. 해결방법을 좀 알 수 있을까요?

**Comment by seolryuhee** [comment]
안녕하세요! 스프링부트와 AWS로 혼자 구현하는 웹 서비스 책 잘 읽고 있습니다.
책을 읽다가 궁금점이 생겨 질문 드립니다.
DB를 postgresql로 해서 JPA Auditing을 통해 CreatedDate를 구현하니
2020-01-09T15:00:07.496 이런 식으로 날짜 형식이 나오는데 
BaseTimeEntity 쪽이나 다른 쪽을 수정하여 YYYY-MM-DD HH:mm:ss 형식으로
나오게 하는 방법이 있을까요?
인터넷에 찾아봐도 딱히 나오는 것이 없어서 자문 구합니다ㅠㅠ

**Comment by knockknockyoo** [comment]
너무 잘 정리되었네요. 저도 좀 참고해서 제 블로그에 옮겨적어도 괜찮을까요?

**Comment by hue-leica** [comment]
builder()오류 문제 해결했어요!! 여러분 
build.gradle에 2개의 의존성 추가헤주세요
    annotationProcessor('org.projectlombok:lombok')
    testAnnotationProcessor('org.projectlombok:lombok')
드디어해결했습니다!!

**Comment by hyeonseokhan** [comment]
윗 분 덕분에 빌드 문제 해결했네요 ㅎㅎ 복 많이 받으세요! 그리고 작성자님 덕분에 Spring boot 쉽게 배우고 있습니다! 감사합니다

**Comment by cjsxorud111** [comment]
@neity16 감사합니다

**Comment by seolryuhee** [comment]
@seolryuhee
자문자답입니다. 결국 MembersListResponseDto에서
modifiedDate 변수의 타입을 String 으로 선언하고
this.modifiedDate = entity.getModifiedDate().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")); 으로 데이터값이 들어가게 해서 해결했습니다.
이렇게해도 괜찮은지 몰라 질문했으나 답이 없으신 관계로 그냥 이렇게 했네요...

**Comment by rlska105** [comment]
error: cannot find symbol
        this.modifiedDate = entity.getModifiedDate();

저도 PostsListResponseDto에서 이문제를 똑같이 겪고 있습니다.!

**Comment by jsh5890** [comment]
혹시 h2 db연결 안되시는분 참고하세요
이전 버전의 스프링에서 기본 URL은 jdbc:h2:mem:testdb이며 testdb는 기본적으로 만들어졌습니다. 2.3.0부터 url을 언급하지 않으면 데이터베이스 이름이 자동으로 생성됩니다. 자동 생성 된 데이터베이스 이름은 스프링 로그에서 찾을 수 있습니다.

**Comment by oldaim** [comment]
스프링부트 책을 구매해서 공부하고있는 학생입니다. 책에 있는 내용 그대로 적용했는데 405 오류가 발생합니다 ㅠㅠ 해결 방안 있는지 여쭤봅니다.

**Comment by perm4** [comment]
H2 Console에서 JDBC URL을 jdbc:h2:mem:testdb로 수정해도 안됩니다.

jdbc:h2:mem:testdb로 수정하기 전 jdbc:h2:~/test 이거로 해도 안되구요

어떻게 해야하죠? 도와주세요..

제 소스코드는 깃헙에 올라가 있습니다.

https://github.com/perm4/spring-webservice

**Comment by seogineer** [comment]
H2 console에서 JDBC URL이 기본  jdbc:h2:~/test 이렇게 되어 있는데
스프링 실행하실 때 로그 잘 보시면 아래와 같은 로그가 있습니다.
이 로그에서 "jdbc ~ e0" 까지 복사하셔서 JDBC URL에 넣어보세요.

2021-03-22 20:19:12.929  INFO 21120 --- [           main] o.s.b.a.h2.H2ConsoleAutoConfiguration    : H2 console available at '/h2-console'. Database available at 'jdbc:h2:mem:a585eee9-aba1-4162-9d5e-392e8085f7e0'

**Comment by dbswhd4932** [comment]
너무 좋은글감사합니다.
백엔드 개발자를 준비하고있는데 천천히 이해하면서 따라가 보겠습니다!

**Comment by kimgun95** [comment]
위 모든 과정을 처음엔 성공적으로 실행시켰는데 다음날 다시 실행해보니 Entity가 table에 매핑이 안되는 분들! 있으실 겁니다.
url 설정 제대로 되었고, postman으로 요청시 응답도 ok를 받는데 왜 테이블 매핑이 안되는 걸까? 그 결과,,,
localhost:8080/h2-console 접속시  port 번호 확인하시길 바랍니다.
저는 뜬금없이 8082로 실행되고 있더라고요... 그래서 8080으로 바꾸니 테이블 확인을 할 수 있었습니다.
하,, 해결해서 다행이다.
